src/gui/widgets/FxLine.cpp

```diff
@@ -157,7 +180,13 @@ void FxLine::drawFxLine( QPainter* p, const FxLine *fxLine, bool isActive, bool
 	int width = fxLine->rect().width();
 	int height = fxLine->rect().height();
 
-	p->fillRect( fxLine->rect(), isActive ? fxLine->backgroundActive() : p->background() );
+	QColor color = muted ? ( isActive ? fxLine->backgroundActive().color() : p->background().color() )
```

Comment:
Still hard to read, there's tons of nesting without much visual help to see it. I think separate functions is the way to go. If you pass the channel in as an argument, you could make them static with const arguments:

```
static QColor activeColor( const FxChannel channel )
{
    //Some logic
}
```
and the same signature for inactiveColor. That should make them pure, which is nice.

Reply:
Okay, I'll break this one out into functions.